### ISSUE TITLE
Incorrect `reportUninitializedInstanceVariable` behavior for classes derived from a Protocol

### ISSUE THREAD

**Issue by rshanor** [microsoft/pyright#10915]
**Describe the bug**
Incorrect `reportUninitializedInstanceVariable` behavior for classes derived from a Protocol 

**Code or Screenshots**
```python
from typing import Protocol

class FooProtocol(Protocol):
    def foo(self) -> None:
        ...


class Foo:
    def __init__(self) -> None:
        pass

    def foo(self) -> None:
        self._foo = 10  # This correctly reports reportUninitializedInstanceVariable


class FooP(FooProtocol):
    def __init__(self) -> None:
        pass

    def foo(self) -> None:
        self._foo = 10  # This does not report an error
```

**Comment by erictraut** [comment]
This is by design. Abstract classes (including classes that explicitly derive from a protocol) are exempt from this check unless they're marked `@final`. See [this issue](https://github.com/microsoft/pyright/issues/5009) for details.
